### ISSUE TITLE
Allow to create user defined LavalinkTrack

### ISSUE THREAD

**Issue by SKProCH** [angelobreuer/Lavalink4NET#97]
*After recent changes, it became difficult to update my fork library (because a big difference between the codebase has accumulated) and I want to port some of my changes to upstream, which would give up using the fork.*

**Is your feature request related to a problem? Please describe.**
In my bot i have a custom resolvers for non usual sources (like a spotify or Yandex Music). This sources cant be resolved as tracks via lavalink itself. I need approach to create user defined tracks (i have all track info - author,  duration, isLiveStream, source, etc).
Main problem - is to form correct identifier. 
In fact identified is a just properly encoded track info.
Also, i need some ghost tracks - these are tracks that we have information about, but were not recognized through Lavalink (to save ratelimit or due to temporary links to the source). For this tracks i need to resolve it only when it actually start playing.

**Describe the solution you'd like**
- In my implementation i created a method what allows to encode track info to identifier
- I add public ctor to LavalinkTrackInfo which contains all properties, to allow creating LavalinkTrackInfo from user code
- Add `Task<LavalinkTrack> GetPlayableTrack()` method to LavalinkTrack with base implementation:
```csharp
        public virtual Task<LavalinkTrack> GetPlayableTrack() {
            return Task.FromResult(this);
        }
```
- Change `PlayAsync` method in `LavalinkPlayer` to
```csharp
public virtual async Task PlayAsync(LavalinkTrack track, TimeSpan? startTime = null,
    TimeSpan? endTime = null, bool noReplace = false)
{
    ... 
    var playableTrack = await track.GetPlayableTrack();
    await LavalinkSocket.SendPayloadAsync(new PlayerPlayPayload(GuildId, playableTrack.Identifier, startTime, endTime, noReplace));
    ...
}
```
This allows us to actually resolve (or get/refresh from cache in my case) track url and info.

**Describe alternatives you've considered**
May be will be best to allow creating LavalinkTrack without identifier (just pass LavalinkTrackInfo or LavalinkTrackInfo's content) and form identified according this info. Cuz currently if i have all track info i still need firstly encode it to identifier and secondly pass along with identifier (`TrackDecoder.EncodeTrack(myInfo), myInfo`). 

**Additional context**
- Initial track encoder implementation [here](https://gitlab.com/enlivenbot/Lavalink4NET/-/commit/4fbe81987b5fa42dad3c034c2b4cf33507458c5f)
- Optimized track encoding [here](https://gitlab.com/enlivenbot/Lavalink4NET/-/commit/8c64094881ae1cf32cee577403907a9d13ddadbf)
- Final improvements of track encoding [here](https://gitlab.com/enlivenbot/Lavalink4NET/-/commit/e299b9a4250b9b51abfb50138cb1cc607449f143)
- Added additional methods to TrackDecoder [here](https://gitlab.com/enlivenbot/Lavalink4NET/-/commit/34268a33feb36e4a9b1054e15b92909fc782497e)
- Allow overriding track before sending to player [here](https://gitlab.com/enlivenbot/Lavalink4NET/-/commit/d16dcda0214a7d07878ffa81b5b256d3f3889400)

How it works for me? I have an [YandexMusicResolver](https://gitlab.com/enlivenbot/enliven/-/blob/master/Enliven/Music/Yandex/YandexMusicResolver.cs) which resolves track to [YandexLavalinkTrack](https://gitlab.com/enlivenbot/enliven/-/blob/master/Enliven/Music/Yandex/YandexLavalinkTrack.cs) (inherited from LavalinkTrack). All conversion from YM track to LL track done in ctor:
```csharp
public YandexLavalinkTrack(YandexMusicTrack relatedYandexTrack, IYandexMusicDirectUrlLoader directUrlLoader)
    : base(TrackDecoder.EncodeTrack(relatedYandexTrack.ToTrackInfo()), relatedYandexTrack.ToTrackInfo()) ...
```
Here we converting YM track to LavalinkTrackInfo and encode it to identified. 
Also in YandexLavalinkTrack we have overridden `GetPlayableTrack` method:
```csharp
public override async Task<LavalinkTrack> GetPlayableTrack() {
    var directUrl = await GetDirectUrl(RelatedYandexTrack.Id);
    var newTrackInfo = new LavalinkTrackInfo(Author, Duration, IsLiveStream, IsSeekable, Position, directUrl, Title, directUrl);
    return new LavalinkTrack(TrackDecoder.EncodeTrack(newTrackInfo, StreamProvider.Http, writer => {
        writer.WriteString("mp3");
    }), newTrackInfo);
}
```
This method refreshes (or get) direct url to track (in YM direct url is non constant and expire in ~10 minutes) and construct an actual track which will be passed to lavalink (but for us we still have initial `YandexLavalinkTrack` in `CurrentTrack`, not temp track we created in `GetPlayableTrack`). It is also worth noting that we do not give the Lavalink load in the form of track resolving (since we have a direct url), and immediately start playing it.

--------

I also understand that all this is very difficult to understand, but this system works fine for me.

**Comment by angelobreuer** [comment]
I like the first solution. You can make a pull request if you want. I would propose the API as the following:

```csharp
 public virtual ValueTask<LavalinkTrack> GetPlayableTrack() {
    return new ValueTask<LavalinkTrack>(this);
}
```

99% of the requests to this method would be the default call which is called synchronously, so there is no need for using a task.

**Comment by angelobreuer** [comment]
Closed by #99
